File: C++/Level1/Numeric String and English Words.cpp
/*
    https://programmers.co.kr/learn/courses/30/lessons/81301?language=cpp
*/
#include <string>
#include <vector>

using namespace std;

const vector<pair<string, string>> TRANSLATE = {
    {"zero" ,"0"},
    {"one"  ,"1"},
    {"two"  ,"2"},
    {"three","3"},
    {"four" ,"4"},
    {"five" ,"5"},
    {"six"  ,"6"},
    {"seven","7"},
    {"eight","8"},
    {"nine" ,"9"}
};

int solution(string s) {

    for (const pair<string, string>& translate : TRANSLATE)
    {
        const size_t leng = translate.first.length();
        size_t pos = s.find(translate.first);

        while (pos != string::npos)
        {
            s.replace(pos, leng, translate.second);
            pos = s.find(translate.first);
        }
    }
    
    return stoi(s);
}

File: Java/Level1/numeric string and English words.java
/*
	https://school.programmers.co.kr/learn/courses/30/lessons/81301
*/

import java.util.HashMap;
import java.util.Map;

class Solution {
    public int solution(String s) {
        Map<String, String> dict = new HashMap<String,String>() {{
            put("zero" ,"0");
            put("one"  ,"1");
            put("two"  ,"2");
            put("three","3");
            put("four" ,"4");
            put("five" ,"5");
            put("six"  ,"6");
            put("seven","7");
            put("eight","8");
            put("nine" ,"9");
        }};  
         
        char[] charArr = s.toCharArray();

        String alpha = "";
        String result = "";

        for(int i = 0; i < charArr.length; ++i)
        {
            if(Character.isDigit(charArr[i]))
            {
                result += charArr[i];
                continue;
            }

            alpha += charArr[i];
            String val = dict.get(alpha);

            if(val != null)
            {
                result += val;
                alpha = "";
            }
        }

        return Integer.parseInt(result);
    }
}